src/engine.rs

```diff
@@ -966,6 +966,90 @@ mod tests {
         assert_eq!(will_force_compact[0], 1);
     }
 
+    #[test]
+    fn test_purge_trigger_force_rewrite() {
+        let dir = tempfile::Builder::new()
+            .prefix("test_purge_trigger_force_write")
+            .tempdir()
+            .unwrap();
+        let cfg = Config {
+            dir: dir.path().to_str().unwrap().to_owned(),
+            target_file_size: ReadableSize::kb(1),
+            purge_threshold: ReadableSize::kb(10),
+            ..Default::default()
+        };
+
+        let engine =
+            RaftLogEngine::open_with_file_system(cfg, Arc::new(ObfuscatedFileSystem::new()))
+                .unwrap();
+        let data = vec![b'x'; 1024];
+        // write 50 small entries into region 1~3, it should trigger force compact
+        // report.
+        for rid in 1..4 {
+            for index in 0..50 {
+                engine.append(rid, index, index + 1, Some(&data[..20]));
```

Comment:
why use small entry?

Reply:
sparse region will trigger rewrite instead of compact.

Reply:
But `sparse` is determined by count of entries, not size of entries.

Reply:
Yes. small entries is used for the convenience of triggering compact and rewrite as the total log size is smaller. I want each region takes about one log file.